src/plugins/intel_npu/src/compiler/include/graph_transformations.hpp

```diff
@@ -14,9 +17,46 @@
  */
 namespace intel_npu::driverCompilerAdapter {
 
-/**
- * @brief Serialize OpenVINO model to IR
- */
-IR serializeToIR(const std::shared_ptr<const ov::Model>& model, uint32_t supportedVersionByCompiler = 7);
+class IR {
+public:
+    IR(const std::shared_ptr<const ov::Model>& origModel,
+       uint32_t supportedVersionByCompiler = 11,
+       bool largeModel = false);
+
+    std::istream& getXml() {
+        if (!_isLargeModel) {
+            return _xml;
+        } else {
+            return _xmlStream;
+        }
+    }
+
+    std::istream& getWeights() {
+        if (!_isLargeModel) {
+            return _weights;
+        } else {
+            return _weightsStream;
+        }
+    }
+
+private:
+    /**
+     * @brief Serialize OpenVINO model to IR
+     */
+    void serializeToIR(uint32_t supportedVersionByCompiler);
```

Comment:
Not sure this is a good API. I call IR.serializeToIR(version) it's not clear where it's serialized to.

Reply:
Renamed to serializeOVModelToIR(), this is private function and we will not call it outside, hope this can bring less confusion.